### PR TITLE
Revise isAndroid test

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
I'm not sure of the right fix.  But putting up this strawman because it's awkward code that needs to handle 4 environments

1) Deployed JVM
2) Deployed Android
3) IDE
4) Gradle

And it's failing in option 3 above.

File: okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.kt [hunk 1]
```diff
@@ -194,22 +194,26 @@ class AndroidPlatform : Platform() {
     val isAndroid: Boolean = try {
       // Trigger an early exception over a fatal error, prefer a RuntimeException over Error.
       Class.forName("com.android.org.conscrypt.OpenSSLSocketImpl")
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.kt]: Should catch deployed environments here.  I'll add comments.

File: okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.kt [hunk 2]
```diff
@@ -194,22 +194,26 @@ class AndroidPlatform : Platform() {
...
+
+      // account for android-all
+      check(Build.VERSION.SDK_INT > 0)
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.kt]: Should fail before accidentally running in Android mode on JVM.  Defaults to 0 internally.

**Comment by yschimke** [comment]
One option is to check for JVM class and short circuit e.g. sun.security.ssl.SSLContextImpl?

But overall I expect this issue to be annoying until we fix properly.

**Comment by swankjesse** [comment]
Seems reasonable enough.

**Pull request merged**